lib/arjdbc/jdbc/adapter.rb

```diff
@@ -358,6 +358,10 @@ def update_lob_value(record, column, value)
       # @return [Integer, NilClass]
       def last_inserted_id(result)
         if result.is_a?(Hash) || result.is_a?(ActiveRecord::Result)
+          # If table does not have primary key defined
+          return nil if result.first.blank?
+
```

Comment:
:scissors: 
did you review if any one the (MRI) adapters do a similar thing?
cause if not we might be approaching this from the wrong end.
also if we can avoid using `blank?` here - since its not clear whether its and empty `result.first` Hash or a `nil`

Reply:
@kares @rsov If this is only captures test issues with a single adapter I would also prefer this is not in generic part of the code base.  Tough if you are only examining a single db but unless one of the failing CI tests goes green or we can tell there is missing coverage amongst all the adapters....we should probably try and isolate.  Otherwise in 2-3 years we won't know why this line is there.

Reply:
@kares @enebo I can move this into the db2/as400 code. That is probably a much better way of doing this

Reply:
@rsov for me it will lower the barrier to acceptance quite a bit.  I may still comment but I am much more open to you making progress you are happy with than polishing an unreleased/unsupported diamond.

OTOH if you do happen to notice a lack of coverage in AR/AR-JDBC and found a general problem then please go through the extra trouble of putting it in the generic place.

Reply:
Unless I'm missing something, this implementation of `last_inserted_id` is not used by the "big three". SQLite has its own implementation, PG uses the plain AR implementation, adds something on top, MySQL uses plain AR.

@rsov I wonder if it would work for you just deleting that method from `lib/arjdbc/jdbc/adapter.rb` and rely only on the AR provided method?